packages/frontend/src/app/main/repo-list/repo-list.component.spec.ts

```diff
@@ -58,40 +72,25 @@ describe('RepoListComponent', () => {
     fixture = TestBed.createComponent(RepoListComponent);
     component = fixture.componentInstance;
     fixture.autoDetectChanges(true);
-    component.paginator = {
-      firstPage: () => {
-        component.updatePage({
-          pageIndex: 0,
-          pageSize: component.pageSize,
-        } as PageEvent);
-      },
-    } as MatPaginator;
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
-  it('should update the rendered pages on input change', () => {
-    component.pageIndex = 1;
-    const expectedPageSize: number = component.pageSize;
+  it('should update the rendered pages on input change', fakeAsync(() => {
+    component.viewConfig.pageIndex = 1;
+    const expectedPageSize: number = component.viewConfig.pageSize;
+    COMServiceMock.fetchRepositories = () => of(mockRepositories);
 
-    component.repositories = mockRepositories;
+    component.update([], '', '');
+    tick();
+    fixture.detectChanges();
 
-    expect(component._elements).toEqual(mockRepositories);
+    expect(component.viewConfig.elements).toEqual(mockRepositories.repos);
     // reset the index
-    expect(component.pageIndex).toEqual(0);
+    expect(component.viewConfig.pageIndex).toEqual(0);
     // didn't change the page size
-    expect(component.pageSize).toEqual(expectedPageSize);
-  });
-
-  it('should not return to the first page when the paginator is not ready', () => {
-    const expectedPageIndex = 1;
-    component.paginator = undefined;
-    component.pageIndex = expectedPageIndex;
-
-    component.repositories = mockRepositories;
-
-    expect(component.pageIndex).toEqual(expectedPageIndex);
-  });
```

Comment:
The "first" page is no longer relevant since we are fetching each page from the API.